pkg/controller/podautoscaler/metrics/legacy_metrics_client.go

```diff
@@ -73,14 +73,14 @@ func (h *HeapsterMetricsClient) GetResourceMetric(resource v1.ResourceName, name
 		return nil, time.Time{}, fmt.Errorf("failed to get pod resource metrics: %v", err)
 	}
 
-	klog.V(4).Infof("Heapster metrics result: %s", string(resultRaw))
-
 	metrics := metricsapi.PodMetricsList{}
 	err = json.Unmarshal(resultRaw, &metrics)
 	if err != nil {
-		return nil, time.Time{}, fmt.Errorf("failed to unmarshal heapster response: %v", err)
+		return nil, time.Time{}, fmt.Errorf("failed to unmarshal heapster response:\n%s\nerror: %v", string(resultRaw), err)
```

Comment:
I'm a bit hesitant about putting the full response in the error message.  I think having the error stay the same and just increasing the log level is probably the safest option here (also, please move the log line back up to where it was, so that we can see the raw response if we need to).

Reply:
> I'm a bit hesitant about putting the full response in the error message. I think having the error stay the same and just increasing the log level is probably the safest option here (also, please move the log line back up to where it was, so that we can see the raw response if we need to).

Thank you, I have done  as you advised.